README.md

```diff
@@ -2,39 +2,106 @@
 
 A collection of composite Github Actions
 
-## terms-of-service-acceptance/run
+## Publish Build Scans® from forked repositories
 
-A composite action to verify that Develocity Terms of Service have been accepted.
+### Description
+When submitting a pull request, a Github workflow that validates the change is usually triggered, however the Develocity Build Scans® can’t be published for 2 reasons:
+- The Develocity Terms of Service have not been agreed to by the contributor
+- Workflows from forked repositories do not have access to secrets although an access token is required to publish a Build Scan®
 
-This action checks if the user submitting a pull-request has accepted the Terms of Service.
-If the user has previously accepted the terms, then the action succeeds.
-If not, a comment is made on the pull-request asking the user to accept and the action fails. The user can then accept the Terms of Service by responding with a specific comment on the pull-request.
+This repository contains some actions which can be combined together to solve this.
 
-See the [cla-assistant-lite documentation](https://github.com/marketplace/actions/cla-assistant-lite) for more details.
+### Architecture
+![Architecture](./doc/architecture.png)
+
+### Usage
+
+**Usage**:
+
+Insert the `Save Build Scan` step after each Maven execution step in the Github workflow called to validate a pull-request (`Build with Maven` here).
+
+```yaml
+[...]
+      - name: Build with Maven
+        run: mvn clean package
+      - name: Save Build Scan
+        uses: gradle/github-actions/maven-build-scan/save@v1.0
+[...]
+```
+
+Add a workflow to publish the Build Scans® saved during the previous step
+
+_Note:_
+Some parameters need to be adjusted here:
+- The workflow name (here `Build`) triggered when a pull-request is submitted
+- The build workflow filename (here `build.yml`) has to be adjusted to the filename of the workflow using `maven-build-scan/save`
+- The location of the Develocity Terms of Service (here `https://foo.bar/tos.html`)
+- The Develocity URL (here `https://<MY_DEVELOCITY_URL>`)
+- The secret name holding the Develocity access key (here `<DEVELOCITY_ACCESS_KEY>`)
+
+```yaml
+name: Upload Build Scans
+
+on:
+  workflow_run:
+    workflows: [ "Build" ]
+    types: [ completed ]
+  issue_comment:
+    types: [ created ]
+
+jobs:
+
+  publish-build-scans:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Publish Build Scan
+        uses: gradle/github-actions/maven-build-scan@v1.0
+        with:
+          build-workflow-filename: 'build.yml'
+          tos-location: 'https://foo.bar/tos.html'
+          develocity-url: 'https://<MY_DEVELOCITY_URL>'
+          develocity-access-key: ${{ secrets.<DEVELOCITY_ACCESS_KEY> }}
+```
```

Comment:
YES! This is all so much simpler from the user point of view, and will make it easier for us to evolve the implementation under the covers.